src/dbnode/storage/entry.go

```diff
@@ -151,6 +151,26 @@ func (entry *Entry) IndexedRange() (xtime.UnixNano, xtime.UnixNano) {
 	return min, max
 }
 
+// ReconciledOnIndexSeries attempts to retrieve the most recent index entry from the
+// shard if the entry this method was called on was never inserted there. If there
+// is an error during retrieval, simply returns the current entry. Additionally,
+// returns a cleanup function to run once finished using the reconciled entry and
+// a boolean value indicating whether the result came from reconciliation or not.
+func (entry *Entry) ReconciledOnIndexSeries() (doc.OnIndexSeries, doc.ReconciledOnIndexSeriesCleanupFn, bool) {
```

Comment:
Maybe instead of a func + bool, we just return a func and caller checks if it is non-nil?

Reply:
Or always return a func and caller always calls close (and something that func is just noop)

Reply:
The bool is just so we can know if we need to increment the reconciled metric. The caller always calls cleanup which is a noop if there's nothing to reconcile.

Reply:
It is a bit of a Go anti-pattern to check for nilability instead of just returning a bool as the last arg (since most folks will know based on Go APIs to check a bool if it's returned, where as they might not know whether to check if something is nil since it's not a common pattern).